### ISSUE TITLE
[Data] Build PSEM-STRATEGY-DATA-v2 with tight AMI/AliMeeting activity references for #76

### ISSUE THREAD

**Issue by kapitalismho** [kapitalismho/PuriPuly-heart#86]
## Decision

This issue is the new **hard data prerequisite** for #76.

Do not start any official/material #76 training run from `PSEM-STRATEGY-DATA-v1`. Keep that freeze immutable and auditable, but treat it as **superseded for model selection in #76** because its temporal activity authority is not defensible at the 50–200 ms topology scale used by PSEM.

This issue must produce:

```text
label contract: psem-handoff-v1
dataset freeze: PSEM-STRATEGY-DATA-v2
```

The model arms, 3 s evidence geometry, two seeds, common head/losses, untouched EVAL discipline, threshold sweeps, and anti-leakage controls in #76 remain unchanged.

This issue deliberately does **not** create a project-specific human gold boundary set and does **not** independently re-adjudicate the acoustic accuracy of the upstream forced-aligned labels. The released, commit-pinned external reference is adopted as the common temporal authority for all #76 arms. Our responsibility in this prerequisite is reproducible acquisition, correct mapping, deterministic transformation, conservative masking of known annotation-risk cases, leakage-safe splitting, and fail-closed freezing.

---

## Why v1 must not be used for the official gate

The v1 engineering is valuable and should be reused: raw-audio provenance, licenses, identity graphs, historical-exposure reconstruction, WavLM overlap audit, connected-component splitting, deterministic topology generation, tests, census, and fail-closed freezing.

The weak layer is narrower but material:

- the AMI parser currently treats each full `transcriber_start`/`transcriber_end` span from `*.segments.xml` as literal speaker activity;
- the AliMeeting parser currently treats every non-empty participant TextGrid interval as literal speaker activity;
- those spans are then swept directly into silence/singleton/overlap intervals, so loose boundaries can change not only event time but also `direct` vs `gap` vs `overlap`, reliable-solo eligibility, takeover/return identity, short-backchannel episodes, and masking;
- AMI's transcription manual explicitly describes segment boundaries as transcription-management units and asks for roughly 1/4–1/2 s of silence padding when possible;
- a concrete selected v1 meeting, `IS1009a`, contains an AMI A-speaker segment from `54.928` to `60.880` while the released forced alignment resolves several speech islands and internal pauses inside that span;
- Horiguchi et al., ASRU 2025, report large reference disagreement for AMI and AliMeeting and show that a normal evaluation collar does not repair errors caused by padded segments and internal pauses.

References:

- AMI transcription manual: https://groups.inf.ed.ac.uk/ami/corpus/Guidelines/speech-transcription-manual.v1.2.pdf
- ASRU 2025 paper: https://arxiv.org/abs/2507.09226
- released forced-aligned labels: https://github.com/nttcslab-sp/diar-forced-alignment
- modern usage precedent: https://huggingface.co/nvidia/diar_streaming_sortformer_4spk-v2.1
- DIHARD III annotation/evaluation plan, as a tight-boundary methodological reference: https://dihardchallenge.github.io/dihard3/

This is not a claim that the corpora, speakers, audio, split work, or PSEM task are bad. It is a requirement to replace the temporal source layer before it is used as ground truth for a strategy decision.

---

# 1. Fixed corpus scope for the #76 gate

## 1.1 Retain

Retain the current source-audio inventory and reusable metadata for:

```text
AMI:        all 68 accepted v1 meetings
AliMeeting: all 8 current Eval_Ali meetings
```

Continue to use as model input:

```text
AMI:        original 16 kHz mono Mix-Headset waveform
AliMeeting: far-field array channel 0 materialized as 16 kHz mono PCM16
```

Synchronized per-participant headset/near-field channels may be used only where needed to reproduce or resolve corpus metadata/reference identities. They are not model inputs and there is no project-specific human listening/audit requirement in this issue.

## 1.2 Add a modest AliMeeting Train tranche

Add a deterministic, whole-component tranche from `Train_Ali`:

```text
nominal target:              16 meetings
allowed component-safe range: 14–18 meetings
target scored exposure:      7–9 hours
```

The purpose is to reduce the current ~89% AMI time dominance and give the English/headset-mix and Mandarin/far-field conditions enough independent meetings for corpus-stratified evaluation, without turning #76 into the later hundred-hour scaling experiment.

Selection must be annotation/model-score blind:

1. acquire the complete Train_Ali metadata needed to resolve meeting, speaker, room/recording-group, waveform, and annotation identities;
2. construct connected components from every knowable shared identity, including recurring speakers and any explicit session/series relationship;
3. enumerate subsets of whole components in the 14–18 meeting range and 7–9 hour range;
4. choose lexicographically by:
   1. maximum coverage of participant-count buckets `{2, 3, 4}`;
   2. maximum number of distinct room/recording-group identities available in official metadata;
   3. minimum absolute distance from 8.0 scored hours;
   4. minimum SHA-256 tie-break value of the sorted component-ID list using salt `PSEM-STRATEGY-DATA-v2`;
5. record the complete candidate census, objective values, selected IDs, and selection hash.

Do not use PSEM topology counts, audio-derived embeddings, VAD predictions, diarization scores, model scores, or #76 outcomes to choose this tranche.

## 1.3 Do not add a third corpus before #76

Do **not** add AISHELL-4, ICSI, CHiME-6, NOTSOFAR-1, DIHARD, DISPLACE, MISP, VoxConverse, or MSDWild to the official #76 TRAIN/DEV/EVAL freeze in this issue.

Reasons:

- AISHELL-4 is attractive but would require a separate temporal-reference decision before being comparable at PSEM's boundary scale;
- ICSI has useful English meeting diversity but introduces another annotation convention that should be evaluated in the later scaling program;
- CHiME-6 is behaviorally valuable and has forced-alignment resources, but introduces a new dinner-party/far-field domain and should be added later;
- NOTSOFAR-1 is a strong modern OOD candidate, but its versioned train/dev transcription state and speaker overlap require separate treatment;
- DIHARD is a methodological reference and possible external check, not the obvious public core training source;
- the remaining candidates add access, license, multimodal, or domain-comparability work that is not necessary to answer the fast strategy-gate question.

The larger public-data scaling/generalization program is intentionally deferred until #76 reveals the winner/runner-up relationship, corpus interaction, topology failures, uncertainty, and compute profile.

---

# 2. Authoritative temporal reference

Pin this exact upstream revision:

```text
repository: https://github.com/nttcslab-sp/diar-forced-alignment
commit:     9527b7c64846fb38316a610f32e9d3466bd6d8b7
```

Use the released RTTMs as the canonical lexical/floor-speech boundary authority for every included AMI and AliMeeting session:

```text
AMI/{train,dev,test}/*.rttm
AliMeeting/{Train_Ali_far,Eval_Ali_far}/*.rttm
```

The upstream project is the released artifact for Horiguchi et al., ASRU 2025, and was created with Montreal Forced Aligner using pinned English/Mandarin pretrained alignment resources. The upstream authors explicitly state that the labels are not perfect and may retain alignment errors, transcription ambiguity, and OOV-related errors.

For #76, those limitations are accepted as limitations of the common external reference. **This project does not independently estimate their acoustic boundary error and does not condition dataset acceptance on a new human audit.** All three model arms must consume and be evaluated against exactly the same reference.

Rules:

- hash every consumed RTTM and bind the hash, upstream commit, source license, and original corpus identity into provenance;
- convert decimal RTTM times to exact 16 kHz half-open source-sample intervals with the existing exact-decimal policy;
- union overlapping/adjacent RTTM rows for the same speaker only when they already overlap or touch; do not invent extra speech padding;
- preserve simultaneous rows from different speakers as overlap;
- map RTTM speaker IDs fail-closed to the official corpus speaker identities;
- treat the RTTM as the activity authority, not as a suggestion to be expanded back to the original ASR segment;
- do not run a neural VAD, diarization model, or any other pseudo-labeler to modify ground truth;
- do not reproduce or improve MFA inside this prerequisite. The released, commit-pinned labels are the reproducible input for #76.

The original AMI word/segment XML and AliMeeting TextGrid remain authoritative for transcript text, speaker metadata, source identities, and explicit nonlexical-risk metadata, but **not** for literal speech onset/offset.

The v1 `annotation_boundary_jitter = 50 ms` value remains a deterministic PSEM topology-reconciliation tolerance. It must not be described as an empirical accuracy bound on either the old source annotations or the forced-aligned reference.

---

# 3. PSEM activity and nonlexical semantics

The new contract must distinguish aligned floor-bearing speech from known unresolved vocal/non-speech annotations without starting a separate annotation research project.

## 3.1 Count as floor-speech activity

Count as activity when represented in the pinned RTTM:

- aligned lexical words;
- filled pauses and hesitation words represented as words, such as `uh`, `um`, and equivalents;
- conventional lexicalized backchannels represented as words, such as `mm-hmm` and equivalents;
- truncated or mispronounced spoken words when an aligned interval exists.

These can maintain or briefly take the conversational floor.

## 3.2 Conservatively mask known unaligned human vocalization risk

Do not silently turn an explicitly annotated human vocalization into a clean silence/handoff simply because it is absent from an `only_words` forced alignment.

Use an `ambiguous_nonlexical_vocalization` mask class for handoff/relation supervision where the original corpus provides an explicit human-vocal event that is not localized by the RTTM.

AMI deterministic policy:

- a nonzero-duration `<vocalsound>` interval is masked over its explicit interval plus 50 ms on each side, clipped to the source timeline;
- a point/zero-duration `<vocalsound>` is masked over `[t - 250 ms, t + 250 ms)`;
- punctuation and disfluency markers with no acoustic duration do not create activity;
- unknown human-vocal annotation element types fail preflight until added to a committed mapping table.

AliMeeting deterministic policy:

- a TextGrid interval whose content is entirely a human-vocal/nonlexical marker after normalization is masked over that full interval;
- a mixed lexical-plus-nonlexical interval remains governed by the forced-aligned lexical RTTM for activity, but handoff/relation targets touching the unresolved interval are masked conservatively;
- non-human noise markers do not create speaker activity;
- every distinct markup/token class and its action must appear in a committed inventory; an unseen class fails preflight rather than being guessed.

This masking policy is intentionally conservative and deterministic. There is no human-audit stratum and no requirement to repair or localize these events further in #86.

Preserve the existing numerical PSEM task constants unless annotation-only deterministic diagnostics expose an internal contradiction:

```text
reliable_solo_min_duration      = 200 ms
annotation_boundary_jitter      =  50 ms
gap_topology_min_duration       = 100 ms
overlap_topology_min_duration   = 100 ms
local_continuity_max_gap        = 1200 ms
short_backchannel_min_duration  = 200 ms
short_backchannel_max_duration  = 1000 ms
```

Any numerical change requires a documented `psem-handoff-v1` amendment made without model scores and before the final split/freeze. Do not tune these constants to make the new reference resemble v1.

---

# 4. Required old-vs-new annotation census

For every retained v1 session and every added AliMeeting session, emit machine-readable and human-readable comparisons between:

```text
v1 source reference
v2 forced-aligned reference
```

At minimum report by corpus and meeting:

- speech, silence, overlap, and reliable-solo hours;
- boundary displacement quantiles where a deterministic correspondence exists;
- original speech removed from internal pauses versus outer padding where derivable;
- direct/gap/overlap topology confusion matrix;
- handoff target additions/removals;
- overlap takeover/return changes;
- short-backchannel episode changes;
- masked-transition changes and reasons;
- proportion of v1 topology episodes that retain identity and event time within 50, 100, 200, and 500 ms;
- alignment-risk and nonlexical-mask counts.

This is a **diagnostic migration report, not a quality gate**. Large disagreement with v1 is not itself a failure because correcting v1's temporal authority is the purpose of this issue. No model predictions may participate.

The report must explicitly distinguish timing-only changes from topology-changing changes so future readers can understand how materially the reference migration changed the PSEM dataset.

---

# 5. Engineering integrity checks only

There is no project-specific human validation or independently measured ground-truth accuracy threshold in this issue.

The following checks remain mandatory because they verify that the chosen external reference was consumed correctly rather than judging whether the upstream alignment is acoustically perfect.

Fail preflight if any of the following is violated:

- the exact upstream repository/commit cannot be reproduced;
- a consumed RTTM hash is missing or changes unexpectedly;
- a selected session has no expected RTTM, has an unexpected duplicate reference, or maps to the wrong corpus/session;
- an RTTM speaker ID cannot be mapped fail-closed to an official source speaker identity;
- an RTTM row has non-finite timing, negative start, non-positive duration, or extends outside the accepted scored waveform range without an explicitly documented source-tail rule;
- an unknown human-vocal markup/token class appears without a committed deterministic action;
- canonical timeline generation is nondeterministic;
- topology generation is nondeterministic;
- repeated regeneration from the same bound inputs changes manifest or freeze hashes unexpectedly;
- any existing leakage, prior-exposure, WavLM-overlap, topology-minimum, corpus-balance, or freeze-integrity gate fails.

These checks must be described as **reference integrity / pipeline correctness checks**, not as evidence that the upstream forced-aligned boundaries have been independently human-verified by this project.

Create at minimum:

```text
reference_provenance.json
reference_integrity_report.json
REFERENCE_MIGRATION.md
```

or equivalent deterministic artifacts that are bound into the final dataset freeze.

---

# 6. Split and EVAL contract

Rerun the full connected-component split optimizer after v2 normalization. Preserve all current leakage exclusions and exact-session WavLM pretraining-overlap checks.

Retain all existing #76 hard minima for natural hours, meetings, topology counts, stable singleton exposure, ongoing overlap exposure, and untouched EVAL.

Add these corpus-balance gates:

```text
TRAIN:
  both corpora present
  AliMeeting >= 5.0 scored hours
  no corpus > 80% of scored hours

DEV:
  both corpora present
  each corpus >= 2 meetings and >= 1.0 scored hour
  no corpus > 75% of scored hours

EVAL:
  both corpora present
  each corpus >= 4 meetings and >= 2.0 scored hours
  no corpus > 70% of scored hours
```

All speaker/component, waveform, session-series, historical-selection, and pretraining-overlap exclusions remain hard constraints. If the exact gates are infeasible, do not silently relax them; emit the infeasibility certificate and revise the contract before freezing.

The #76 evaluator must produce, at every shared score threshold:

1. pooled exposure-weighted frontiers;
2. equal-corpus macro frontiers;
3. separate AMI and AliMeeting frontiers;
4. topology-by-corpus slices;
5. corpus-stratified meeting bootstrap intervals.

Do not tune separate thresholds per corpus.

A strategy must not be declared the universal winner from the pooled frontier alone. If the equal-corpus result and corpus-specific results materially reverse the decision, report the #76 outcome as corpus-dependent/inconclusive under the existing decision framework.

---

# 7. Implementation work

Reuse and evolve the existing `experiments/psem_training_strategy_gate/data` pipeline rather than replacing proven infrastructure.

Required implementation:

- add commit-pinned NTT RTTM acquisition/provenance and SHA-256 verification;
- add AMI and AliMeeting RTTM parsers with exact source-sample coordinates;
- add explicit nonlexical-risk inventory and deterministic ambiguity masks;
- acquire/census/select the bounded AliMeeting Train tranche;
- version the operational contract as `psem-handoff-v1`;
- regenerate canonical timelines and all topology labels;
- generate old-vs-new migration/census artifacts;
- generate the reference-integrity/provenance artifacts described above;
- rerun identity components, prior-exposure audit, WavLM overlap audit, split optimization, census, and preflight;
- add unit/property/integration tests for reference parsing, speaker mapping, masks, topology flips, component selection, split balance, hashes, and fail-closed behavior;
- freeze all inputs and derived identities as `PSEM-STRATEGY-DATA-v2`;
- write a completion report and update #76 so the official six material runs are unblocked only from the accepted v2 freeze.

Explicitly **not required**:

- manual listening or human boundary annotation;
- a 300-clip gold audit packet;
- reviewer agreement measurements;
- project-specific boundary-accuracy estimates;
- reproducing or improving MFA;
- fixing upstream alignment errors discovered only through optional inspection;
- selecting a third training corpus;
- training any #76 arm;
- using model predictions to alter dataset/reference decisions.

---

# 8. Freeze and completion conditions

Create a new immutable freeze:

```text
PSEM-STRATEGY-DATA-v2
```

The freeze must bind at least:

- `psem-handoff-v1`;
- exact NTT upstream repository + commit;
- every consumed RTTM hash;
- original waveform/source identities and hashes;
- original corpus metadata/annotation identities used for speaker/nonlexical metadata;
- retained AMI and AliMeeting Eval source identities;
- selected AliMeeting Train tranche and deterministic selection receipt;
- canonical timeline hashes;
- topology-label hashes;
- prior-exposure evidence;
- identity-component graph;
- WavLM overlap audit;
- split assignment;
- old-vs-new migration/census artifacts;
- reference-integrity artifacts;
- preflight result;
- relevant repository code/config hashes.

Completion requires:

```text
[ ] exact upstream reference revision pinned
[ ] every selected source/reference identity resolved
[ ] all reference-integrity checks pass
[ ] v2 canonical timelines/topology labels regenerated deterministically
[ ] old-vs-new migration report generated
[ ] all #76 natural-data/topology minima pass
[ ] corpus-balance gates pass
[ ] leakage/prior-exposure/pretraining-overlap gates pass
[ ] final dataset preflight passes
[ ] PSEM-STRATEGY-DATA-v2 freeze produced
[ ] #76 updated with the accepted v2 completion/freeze reference
```

There is **no human-audit acceptance gate**. The final report must state the limitation plainly:

> `PSEM-STRATEGY-DATA-v2 adopts the commit-pinned forced-aligned AMI/AliMeeting references released by Horiguchi et al. (ASRU 2025) as the common temporal activity reference. This project does not perform additional manual boundary adjudication or independently estimate their acoustic boundary accuracy.`

That limitation applies equally to all three #76 arms and must be carried into the eventual #76 report/publication claims.

---

# 9. Post-#76 scaling remains deferred

Do not create or execute the large public-data scaling/generalization program inside #86.

Create that follow-up only after #76 reports at least:

- arm/seed pooled Pareto frontiers;
- equal-corpus macro frontiers;
- AMI and AliMeeting corpus-specific frontiers;
- topology × corpus error analysis;
- boundary timing analysis;
- meeting-level bootstrap uncertainty;
- training/inference compute and throughput;
- winner / close runner-up / corpus-dependent / inconclusive decision.

At that point the next issue can decide, using actual evidence, whether to:

- scale only a clear winner;
- retain a close runner-up for data-scaling curves;
- investigate corpus/domain interaction first;
- or prioritize topology-specific data/reference work if all strategies remain weak.

Likely candidates for that later program include larger AliMeeting Train coverage, AISHELL-4 after its own reference decision, CHiME-6, ICSI, and a modern OOD panel such as NOTSOFAR-1. Their exact roles and data budgets are intentionally **not** frozen here.
